MonumentAddons.cs

```diff
@@ -3631,6 +3634,213 @@ private interface IDeepCollection
             return false;
         }
 
+        private class DungeonEntranceMapper
+        {
+            private Dictionary<DungeonGridInfo, MonumentInfo> _entranceToMonument = new Dictionary<DungeonGridInfo, MonumentInfo>();
+
+            public void OnServerInitialized()
+            {
+                foreach (var monumentInfo in TerrainMeta.Path.Monuments)
+                {
+                    if (monumentInfo.DungeonEntrance != null)
+                    {
+                        _entranceToMonument[monumentInfo.DungeonEntrance] = monumentInfo;
+                    }
+                }
+            }
+
+            public MonumentInfo GetMonumentFromTunnel(DungeonGridCell dungeonGridCell)
+            {
+                var entrance = TerrainMeta.Path.FindClosest(TerrainMeta.Path.DungeonGridEntrances, dungeonGridCell.transform.position);
+                if (entrance == null)
+                    return null;
+
+                return GetMonumentFromEntrance(entrance);
+            }
+
+            private MonumentInfo GetMonumentFromEntrance(DungeonGridInfo dungeonGridInfo)
+            {
+                MonumentInfo monumentInfo;
+                return _entranceToMonument.TryGetValue(dungeonGridInfo, out monumentInfo)
+                    ? monumentInfo
+                    : null;
+            }
+        }
+
+        private static class PhoneUtils
+        {
+            public const string Tunnel = "FTL";
+            public const string UnderwaterLab = "Underwater Lab";
+            public const string CargoShip = "Cargo Ship";
+
+            private static readonly Regex SplitCamelCaseRegex = new Regex("([a-z])([A-Z])", RegexOptions.Compiled);
+
+            public static void RenameTelephone(PhoneController phone, string phoneName, Vector3 position, string monumentShortName, MonumentAddons plugin) 
+            {
+                if (ShouldAppendCoordinate(monumentShortName, plugin))
+                    phoneName += $" {PhoneController.PositionToGridCoord(position)}";
+
+                phone.PhoneName = phoneName;
+            }
```

Comment:
This code is unused.

Reply:
If there are potential improvements, I dont mind to implement them. I can apply what I learn here in future PRs, so as long as you are not bothered with code reviews, dont mind me rewriting it. I know this code is not ideal, I hacked togheter some idea and it is basically same code moved elsewhere, so yeah, all insights are welcomed and very helpful.